NString.Tests/StringTemplateConverterTests.cs

```diff
@@ -0,0 +1,335 @@
+            public IReadOnlyCollection<string> ConcatenatedStringCollectionField;
+        }
+
+        class ReferenceOrValueTypeValues
+        {
+            [StringTemplateValueConverter(typeof(StringCollectionConverter))]
+            public IReadOnlyCollection<string> ConcatenatedStringCollection { get; set; }
+
+            [StringTemplateValueConverter(typeof(DateTimeToTicksConverter))]
+            public DateTime DateTimeToTicks { get; set; }
+        }
+
+        class ValuesForFormatProvider
+        {
+            [StringTemplateValueConverter(typeof(AddFortyTwoDaysConverter))]
+            public DateTime ConvertedWithFortyTwoDaysAdded { get; set; }
+
+            [StringTemplateValueConverter(typeof(DateTimeToTicksConverter))]
+            public DateTime ConvertedToTicks { get; set; }
+
+            [StringTemplateValueConverter(typeof(GetDayNameConverter))]
+            public DateTime ConvertedToDayName { get; set; }
+        }
+
+        class MiscConvertersValues
+        {
+            [StringTemplateValueConverter(null)]
+            public int Attribute_With_Null_Converter_Type { get; set; }
+
+            [StringTemplateValueConverter(typeof(CanNeverConvertConverter))]
+            public int Converter_CanConvert_Always_False { get; set; }
+
+            [StringTemplateValueConverter(typeof(IncrementStateConverter))]
+            public int State_Mutator_Converter { get; set; }
+
+            [StringTemplateValueConverter(typeof(StringCollectionConverter))]
+            public string[] String_Array_With_ReadOnlyCollection_Converter { get; set; }
+
+            [StringTemplateValueConverter(typeof(NullToBoolConverter))]
+            public object NullToBool { get; set; }
+        }
+
+        class SecondMiscConvertersValues
+        {
+            [StringTemplateValueConverter(typeof(IncrementStateConverter))]
+            public int State_Mutator_Converter2 { get; set; }
+        }
+
+        class EnumValues
+        {
+            [StringTemplateValueConverter(typeof(EnumValuesConverter))]
+            public DayOfWeek Day { get; set; }
+
+            public static string GetCustomDayString(DayOfWeek day)
+            {
+                switch (day)
+                {
+                    case DayOfWeek.Sunday:
+                    case DayOfWeek.Saturday:
+                        return "Week-end! :)";
+                    default:
+                        return $"{day} + :(";
+                }
+            }
+
+            class EnumValuesConverter : StringTemplateValueConverter<DayOfWeek>
+            {
+                public override object Convert(DayOfWeek value)
+                {
+                    switch (value)
+                    {
+                        case DayOfWeek.Sunday:
+                        case DayOfWeek.Saturday:
+                            return "Week-end! :)";
+                        default:
+                            return $"{value} + :(";
+                    }
```

Comment:
Could you call `GetCustomDayString` here so that the logic isn't repeated?

Reply:
oops!